Lombiq.Hosting.Azure.ApplicationInsights/Services/TelemetryFilter.cs

```diff
@@ -0,0 +1,37 @@
+﻿using Microsoft.ApplicationInsights.Channel;
+using Microsoft.ApplicationInsights.DataContracts;
+using Microsoft.ApplicationInsights.Extensibility;
+using System.Collections.Generic;
+
+namespace Lombiq.Hosting.Azure.ApplicationInsights.Services;
+
+public class TelemetryFilter : ITelemetryProcessor
+{
+    private readonly List<string> _errors = new()
+    {
+        "Azure.RequestFailedException: The specified container already exists.",
+        "The specified blob does not exist.",
+        "Microsoft.Data.SqlClient.SqlException (0x80131904): There is already an object named",
```

Comment:
These two might be real errors with e.g. a wrong Blob lookup (including a request to a wrong Media file path) or a faulty migrations class too. Only filter out what are expected errors.

Reply:
That's really challenging, I don't see how I could do that with the current info that's coming here to the filter service. I will check if it is even possible. If you have any idea don't keep it in😄

Reply:
One of the SQL ones contains references to a `prefix_Identifiers` table. The blob one I don't know what you found, but I guess that also contains some telltale information.

Reply:
> One of the SQL ones contains references to a prefix_Identifiers table.

That's right. I did that now. Also I could filter out the last children of the "Azure.RequestFailedException: The specified container already exists." error which is a HTTP PUT error where the container name can be matched against. But its parent doesn't have anything referring to the container or anything that would mean that we are handling this specific case.

Same goes with "The specified blob does not exist.", there is no info and can't add info that could be a filter reference. Because there is nothing to check against if it's a real error or just an existence check.

Also what I found online is that you either filter everything or nothing, you can't filter only the "valid" errors for those.

Reply:
> But its parent doesn't have anything referring to the container or anything that would mean that we are handling this specific case.

Please provide an example of such parents.

Why did you add the "The specified blob does not exist." to begin with, which such telemetry did you notice?

> Also what I found online is that you either filter everything or nothing, you can't filter only the "valid" errors for those.

What do you mean? If there's a piece of telemetry we don't want, we can just filter that out, like only that dependency telemetry. What's all or nothing here?

Reply:
Example for parent-child:
![image](https://github.com/Lombiq/Orchard-Azure-Application-Insights/assets/19938236/8d2b1301-ca26-4a74-9fee-f17166190281)
The BlobContainerClient.Create won't hold any specific information to the container itself. Only its child will contain the container name. BTW I'm not even sure if we can specifically filter out any of these Blob calls when it is just an existence check or not. Because the call with the 409 response is also just an error that tells that the container already exists, you don't know if it is supposed to happen because of an existence check. 

So that's what I'm talking about when I say that it is all or nothing. You either filter out all these Blob create errors with "Azure.RequestFailedException: The specified container already exists." that might contain legit errors that shouldn't happen because it should create a container, but you can't filter for those.

> Why did you add the "The specified blob does not exist." to begin with, which such telemetry did you notice?

You will get these when using MediaTheme with Blob when checking for existence:
https://github.com/Lombiq/Hosting-Media-Theme/blob/dev/Lombiq.Hosting.MediaTheme.Bridge/Services/MediaThemeManager.cs#L84

But the problem here that it might filter those call logs where a 404 is supposed to happen because of a wrong link etc.

Reply:
So I suppose we can't do anything to filter out the blob container exists errors, then? BTW can't we get rid of these by removing `"CreateContainer": true` from appsettings? (Just for prod and staging scenarios, locally it's useful.)

For the Media Theme Blob 404s I created https://lombiq.atlassian.net/browse/NEST-484 BTW. That's specific to Media Theme (and rather DotNest) but we shouldn't blanket ignore Blob 404s.

Reply:
> So I suppose we can't do anything to filter out the blob container exists errors, then?
We can remove all the "blob container exists" errors but that means if it would be a legit one we filter out those as well, so we can't filter it that specifically like you asked in your first review.
Btw I think currently it shouldn't really happen, because _blobContainer.Create is only used in two places where it actually throws these errors. So we can filter these in the generic way as I implemented it first.

> BTW can't we get rid of these by removing "CreateContainer": true from appsettings?

It would remove only the DataProtection container related errors, because Media Blob always does a `CreateIfNotExistsAsync` on startup (IModularTenantEvents.ActivatingAsync in MediaBlobContainerTenantEvents)

Reply:
OK, let's filter out all the "Azure.RequestFailedException: The specified container already exists." telemetry, then, but please comment this in the code.